src/api/EscargotPublic.cpp

```diff
@@ -1235,6 +1235,54 @@ void DebuggerOperationsRef::BreakpointOperations::getStackTrace(DebuggerOperatio
     }
 }
 
+void DebuggerOperationsRef::BreakpointOperations::getScopeChain(uint32_t stateIndex, DebuggerOperationsRef::ScopeChainVector& outScopeChain)
+{
+    outScopeChain.clear();
+
+    ExecutionState* state = toImpl(m_executionState);
+
+    while (stateIndex > 0) {
+        state = state->parent();
+        stateIndex--;
+
+        if (state == nullptr) {
+            return;
+        }
+    }
+
+    if (state->isNativeFunctionObjectExecutionContext()) {
+        outScopeChain.push_back(NATIVE_ENVIRONMENT);
+        return;
```

Comment:
Looks good, but it just returns when it meets a native function scope.
Its possible that native function call is a part of the entire scope chains like a native function call in a normal JS function.
How about iterating on the `ExecutionState` chain?

Reply:
I may misunderstand something. I thought scope works this way:
```
stack[0] -> scope ->scope -> scope -> ...
stack[-1] -> scope ->scope -> scope -> ...
stack[-2] -> scope ->scope -> scope -> ...
...
```

The different scope chains might share scope objects (in fact they do it frequently). Native functions has no scope, so they are not participating when chains are built.

Do native functions have scope objects and they can be part of scope chains built for normal functions?

Reply:
Oh, I was confused between scope(lexical) and call stack.
If you want to get lexical scope chain, then native function should be excluded because it does not have any valid scope.
Sorry to make you confused.

Reply:
Shall I do any changes in the patch?

Reply:
Yes, please remove `NATIVE_ENVIRONMENT` part since its no longer necessary for scope chains.

Reply:
Thank you patch is updated.

Reply:
Btw, shall I change the name to lexicalScopeChain to avoud user confusion?

Reply:
Good idea!
Yes, please change the name of function. Thanks!